crates/autopilot/src/solvable_orders.rs

```diff
@@ -170,29 +174,62 @@ impl SolvableOrdersCache {
     pub async fn update(&self, block: u64) -> Result<()> {
         let start = Instant::now();
         let min_valid_to = now_in_epoch_seconds() + self.min_order_validity_period.as_secs() as u32;
-        let db_solvable_orders = self.persistence.solvable_orders(min_valid_to).await?;
 
-        let mut counter = OrderFilterCounter::new(self.metrics, &db_solvable_orders.orders);
+        // A new auction should be created regardless of whether new solvable orders are
+        // found. The incremental solvable orders cache updater should only be
+        // enabled after the initial full SQL query
+        // (`persistence::solvable_orders`) returned some orders. Until then, `MIN_UTC`
+        // is used to indicate that no orders have been found yet by
+        // (`persistence::solvable_orders`). This prevents situations where
+        // starting the service with a large existing DB would cause
+        // the incremental query to load all unfiltered orders into memory, potentially
+        // leading to OOM issues.
+        let (db_solvable_orders, previous_creation_timestamp) = {
+            let lock = self.cache.lock().await;
+            match &*lock {
+                Some(cache) if cache.last_order_creation_timestamp > DateTime::<Utc>::MIN_UTC => (
+                    self.persistence
+                        .solvable_order_after(
+                            &cache.solvable_orders,
+                            cache.last_order_creation_timestamp,
+                            cache.solvable_orders.latest_settlement_block,
+                            min_valid_to,
+                        )
+                        .await?,
+                    cache.last_order_creation_timestamp,
+                ),
+                _ => (
+                    self.persistence.all_solvable_orders(min_valid_to).await?,
+                    DateTime::<Utc>::MIN_UTC,
+                ),
+            }
+        };
+
+        let latest_creation_timestamp = db_solvable_orders
+            .orders
+            .values()
+            .map(|order| order.metadata.creation_date)
+            .max()
+            .unwrap_or(previous_creation_timestamp);
```

Comment:
Does this actually reduce the latest timestamp if the last created order gets cancelled?
To get around that you could chain `[previous_creation_timestamp]` onto the iterator and always `.uwrap()`.
Or you do `std::cmp::max(max(order.metadata.creation_data),  previous_creation_timestamp)`

Reply:
Good catch, fixed.